client/v2_2/save_.py

```diff
@@ -172,6 +173,21 @@ def write_file(name, accessor,
     with io.open(name, u'wb') as f:
       f.write(accessor(arg))
 
+  def write_file_and_store(name, accessor,
+                           arg, cached_layer):
+    write_file(cached_layer, accessor, arg)
+    link(cached_layer, name)
+
+  def link(source, link):
```

Comment:
Why do you need to unlink if the link exists? Does it mean you always unlink and link?